argparse.c

```diff
@@ -114,7 +121,8 @@ argparse_getvalue(struct argparse *self, const struct argparse_option *opt,
         if (s[0] != '\0')
             argparse_error(self, opt, "expects a numerical value", flags);
         break;
-    default:
+    case ARGPARSE_OPT_END:
+    case ARGPARSE_OPT_GROUP:
```

Comment:
I guess it's better to have a `default` statement here. Even if all enumeration values are covered here, the programmer can still fill arbitrary integer value in the option type field. It's defensive programming.

From my understanding, I don't think it's necessary to fix all warnings emitted with `-Weverything` flag.

Reply:
You are right, of course, and that's why they're warnings and not errors. It's a pull request, you may disapprove it at will. Anyway, filling arbitrary values to an enum of specific type is a kind of misbehaviour you can't save people from, but I guess it always pays off to be a bit more defensive.